src/test/java/io/split/android/engine/splits/SplitCacheTest.java

```diff
@@ -108,5 +112,82 @@ public void testGetSplitNames() {
 
     }
 
+    @Test
+    public void testWriteToDiskConcurrency() throws Exception {
+        final String ROOT_FOLDER = "./build";
+        final String FOLDER = "thefolder";
+        final String CHANGE_NUMBER_FILE = "SPLITIO.changeNumber";
+        final String FILE_PREFIX = "SPLITIO.split.";
+        final String JSON_SPLIT_TEMPLATE = "{\"name\":\"%s\"}";
+
+        File rootFolder = new File(ROOT_FOLDER);
+        IStorage storage = new FileStorage(rootFolder, FOLDER);
+        //IStorage storage = new MemoryStorage();
+        CountDownLatch latch = new CountDownLatch(2);
+
+        for(int i = 0; i < 10000; i++) {
+            String splitName = "split-" + i;
+            try {
+                storage.write(FILE_PREFIX + splitName, String.format(JSON_SPLIT_TEMPLATE, splitName));
```

Comment:
is this `.write` fails this test proceeds. Is that the expected behavior?

Reply:
Nice catch. I've removed. Thank you.